reduce.go

```diff
@@ -0,0 +1,71 @@
+package godash
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// Reduce reduces the given collection using given reduce function
+func Reduce(in, out, reduceFn interface{}) error {
+	input := reflect.ValueOf(in)
+	output := reflect.ValueOf(out)
+	if err := isReferenceType(output); err != nil {
+		return err
+	}
+
+	reducer := reflect.ValueOf(reduceFn)
+	if err := validateReducer(reducer); err != nil {
+		return err
+	}
+
+	if input.Kind() == reflect.Slice {
+		outputKind := output.Elem().Kind()
+		reducerFnType := reducer.Type()
+		if outputKind != reducerFnType.In(0).Kind() {
+			return fmt.Errorf("reducer function's first argument has to be the type of out")
+		}
+		if input.Type().Elem().Kind() != reducerFnType.In(1).Kind() {
+			return fmt.Errorf("reducer function's second argument has to be the type of element of input slice")
```

Comment:
Could you enrich error message with what exact Type is expected similar to filter.go

Reply:
Enriched. Could you please review now?